packages/@aws-cdk/aws-iam/lib/role.ts

```diff
@@ -385,6 +369,19 @@ export class Role extends Resource implements IRole {
   public grantPassRole(identity: IPrincipal) {
     return this.grant(identity, 'iam:PassRole');
   }
+
+  /**
+   * Return a copy of this Role object whose Policies will not be updated
+   *
+   * Use the object returned by this method if you want this Role to be used by
+   * a construct without it automatically updating the Role's Policies.
+   *
+   * If you do, you are responsible for adding the correct statements to the
+   * Role's policies yourself.
+   */
+  public withoutPolicyUpdates(): IRole {
```

Comment:
Going back to our conversation, why can't this be in `IRole` and works both for `Role` and imported roles?

Reply:
https://github.com/aws/aws-cdk/pull/5569#issuecomment-572538268

1. The interface is already starting to get onerous to implement and I don't want to make it worse. There are 2 different parts to it: "what makes an `IRole` a role", and "operations that can be done to any `IRole`, which are the same for every role anyway". This is part of the 2nd, but I'd argue those shouldn't actually be on `IRole` at all.

2. Even if we put it on `IRole`, my gut says the 99% case will be immutable, hence following up the import by calling this method. We want to optimize for the common case, and adding more verbiage seems the opposite. Either we (ALSO?) add a `withPolicyUpdates()` to switch it back on, or we just leave it as import properties for the `fromRoleArn()`. 

So either we make `IRole` *even heavier* or it doesn't need to be there at all.

Reply:
Sure. It’s not that critical.